scripts/issue/807/withdraw_stables.py

```diff
@@ -0,0 +1,87 @@
+from great_ape_safe import GreatApeSafe
+from helpers.addresses import r
+
+
+def main():
+    vault = GreatApeSafe(r.badger_wallets.treasury_vault_multisig)
+    vault.init_aave()
+    vault.init_compound()
+    vault.init_convex()
+    vault.init_aura()
+    vault.init_balancer()
+    vault.init_curve()
+    vault.init_sushi()
+
+    usdt = vault.contract(r.treasury_tokens.USDT)
+    usdc = vault.contract(r.treasury_tokens.USDC)
+    fei = vault.contract(r.treasury_tokens.FEI)
+    frax = vault.contract(r.treasury_tokens.FRAX)
+    dai = vault.contract(r.treasury_tokens.DAI)
+
+    fxs = vault.contract(r.treasury_tokens.FXS)
+    comp = vault.contract(r.treasury_tokens.COMP)
+    weth = vault.contract(r.treasury_tokens.WETH)
+
+    cDAI = vault.contract(r.treasury_tokens.cDAI)
+    bpt_3pool = vault.contract(r.balancer.B_3POOL)
+
+    aave_stables = [usdt, usdc, fei, frax]
+    a_stables = [
+        vault.contract(r.treasury_tokens[f"a{token.symbol()}"])
+        for token in aave_stables
+    ]
+
+    vault.take_snapshot(
+        tokens=aave_stables + a_stables + [dai, fxs, comp, cDAI, bpt_3pool]
+    )
+
+    # CONVEX
+    # https://etherscan.io/tx/0xa084e77b26cae597bd973c3ab91b3045b7fa873d13cbcc48ac58231a3230f756#eventlog (#189)
+    kek_id = "0377F3E93D8BCB691D20E708122E8F9AFD168512D033F5738EB8CF72862E1090"
+    # withdraw -> afrax
+    vault.convex.withdraw_locked(a_stables[-1], kek_id)
+
+    # AAVE
+    # withdraw -> usdt, usdc, fei, frax
+    for token in aave_stables:
+        vault.aave.withdraw_all(token)
+
+    # COMPOUND
+    # withdraw -> dai
+    vault.compound.withdraw_ctoken(cDAI, cDAI.balanceOf(vault))
+    # claim -> COMP
+    vault.compound.claim_all()
+
+    # AURA/BALANCER
+    # withdraw -> staBAL3
+    vault.aura.unstake_all_and_withdraw_all(bpt_3pool)
+    # withdraw -> usdc, dai, usdt
+    vault.balancer.unstake_all_and_withdraw_all([usdc, dai, usdt], pool=bpt_3pool)
```

Comment:
i tried to adjust this to `vault.balancer.unstake_and_withdraw_all_single_asset(dai, pool=bpt_3pool)` to prevent more usdc withdrawals, but the func is bugged. `min_out` needs to be an int, but more importantly the `underlying_out` calcd is incorrect (i get `-105802305896932179024330388404546093.000000`).

lets leave it for now, but try and fix this in another pr. can you set up a ticket for this?

Reply:
agree, will open a ticket for this. in the meantime, this should raise an error. i actually already addressed this earlier in #778 to catch this miscalc (and the others) like [here](https://github.com/Badger-Finance/badger-multisig/blob/155d2268dc0e8dea86f0290b3a349fb7d87074bf/great_ape_safe/ape_api/balancer.py#L412)